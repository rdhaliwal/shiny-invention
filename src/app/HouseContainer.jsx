import React from 'react';
import House from 'app/House';
import {
  Route,
  Link
} from 'react-router-dom'

import HogwartsLogo from 'assets/hogwarts-logo.jpg';

export default class HouseContainer extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    let tabbar = [];
    tabbar = this.props.houseNames.map(h =>
      <div className={`HouseContainer-tab HouseContainer-tab--${h.toLowerCase()}`}
           key={h.toLowerCase()}>
        <Link to={`/app/${h.toLowerCase()}`} className={true ? 'is-selected' : ''}>
          {h}
        </Link>
      </div>
    );

    return (
      <div className="HouseContainer">
        <div className="HouseContainer-header">
          <h1 className="HouseContainer-heading">Hogwarts School of Witchcraft and Wizadry</h1>
          <img src={HogwartsLogo} alt="Hogwarts" className="HouseContainer-logo"/>
        </div>
        <div className="HouseContainer-tabbar">
          {tabbar}
        </div>
        <div className="HouseContainer-viewContainer">
          <Route path={`${this.props.match.url}/:houseName`} component={House}/>
        </div>
      </div>
    );
  }
}
