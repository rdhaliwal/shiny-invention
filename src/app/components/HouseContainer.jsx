import React from 'react';
import House from 'app/components/House';
import {
  Route,
  Link
} from 'react-router-dom'

import HogwartsLogo from 'assets/hogwarts-logo.jpg';

const Tab = ({tabKey, tabName, isActive, additionalClasses}) => (
  <div className={`HouseContainer-tab ${isActive ? 'is-selected' : ''} ${additionalClasses}`}>
    <Link to={`/app/${tabKey}`} className="HouseContainer-tabLink">{tabName}</Link>
  </div>
);

const TabBar = function(houseNames, currentHouse) {
  let bar = houseNames.map(house =>
    <Tab key={house.toLowerCase()}
         tabKey={house.toLowerCase()}
         tabName={house}
         isActive={currentHouse === house.toLowerCase()}
         additionalClasses={`HouseContainer-tab--${house.toLowerCase()}`}/>
  );

  bar.push(<Tab key='default' tabKey='default' tabName='default' isActive={currentHouse === 'default'} /> );
  return bar;
};

export default class HouseContainer extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="HouseContainer">
        <div className="HouseContainer-header">
          <h1 className="HouseContainer-heading">Hogwarts School of Witchcraft and Wizadry</h1>
          <img src={HogwartsLogo} alt="Hogwarts" className="HouseContainer-logo"/>
        </div>
        <div className="HouseContainer-tabbar">
          {TabBar(this.props.houseNames, this.props.location.pathname.replace("/app/", ""))}
        </div>
        <div className="HouseContainer-viewContainer">
          <Route path={`${this.props.match.url}/:houseName`} component={House}/>
        </div>
      </div>
    );
  }
}
