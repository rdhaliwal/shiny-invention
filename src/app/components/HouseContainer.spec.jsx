// Link.react-test.js
import React from 'react';
import { HouseContainer } from './HouseContainer';
import renderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';

jest.mock('react-router-dom');
jest.mock('app/components/House', () => require('mockComponent'));

// Snapshot test
test('It renders the HouseContainer', () => {
  let props = {
    houseNames: ["one", "two", "three"],
    students: ["Harry"],
    location: {pathname: "/app/three"},
    match: {url : "/app/"}
  }

  const component = renderer.create(
    <HouseContainer {...props} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


// Functionality Tests
test('It fires an action when clicked', () => {
  let props = {houseNames: ["one", "two", "three"],
    students: ["Harry"],
    location: {pathname: "/app/three"},
    match: {url : "/app/"}
  }
  const addStudentClick = jest.fn();
  const wrapper = mount(
    <HouseContainer {...props} preferredWayToAddStudentClick={addStudentClick} />
  );

  const button = wrapper.ref('addStudent');
  button.simulate('click');
  expect(addStudentClick).toHaveBeenCalled();
});

