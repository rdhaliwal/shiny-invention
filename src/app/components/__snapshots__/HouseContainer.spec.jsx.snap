// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders the HouseContainer 1`] = `
<div
  className="HouseContainer"
>
  <div
    className="HouseContainer-header"
  >
    <h1
      className="HouseContainer-heading"
    >
      Hogwarts:
      <br />
      School of Witchcraft and Wizadry
    </h1>
    <img
      alt="Hogwarts"
      className="HouseContainer-logo"
      src="test-file-stub"
    />
  </div>
  <div
    className="HouseContainer-tabbar"
  >
    <div
      className="HouseContainer-tab  HouseContainer-tab--one"
    />
    <div
      className="HouseContainer-tab  HouseContainer-tab--two"
    />
    <div
      className="HouseContainer-tab is-selected HouseContainer-tab--three"
    />
    <div
      className="HouseContainer-tab  undefined"
    />
  </div>
  <div
    className="HouseContainer-viewContainer"
  />
  <div
    className="HouseContainer-studentList"
  >
    Student List:
    <ul>
      <li>
        Harry
      </li>
    </ul>
    <button
      onClick={undefined}
    >
      Add
    </button>
  </div>
</div>
`;
